cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/lineage/field/FieldLineageDatasetTest.java

```diff
@@ -0,0 +1,180 @@
+
+      expectedEndPointField = new EndPointField(destination, "name");
+      actualEndPointFields = fieldLineageDataset.getOutgoingSummary(new EndPointField(source, "body"), 0, 10000);
+      Assert.assertEquals(expectedEndPointField, actualEndPointFields.iterator().next());
+
+      Set<ProgramRunOperations> incomingOperations = fieldLineageDataset.getIncomingOperations(destination, 0, 10000);
+      Set<ProgramRunOperations> outgoingOperations = fieldLineageDataset.getOutgoingOperations(source, 0, 10000);
+      Assert.assertEquals(1, incomingOperations.size());
+      Assert.assertEquals(incomingOperations, outgoingOperations);
+
+      ProgramRunOperations programRunOperations = incomingOperations.iterator().next();
+      Assert.assertEquals(programRunOperations.getChecksum(), fieldLineageInfo.getChecksum());
+
+      Assert.assertEquals(Collections.singleton(programRun1), programRunOperations.getProgramRunIds());
+
+      // test with bigger time range
+      incomingOperations = fieldLineageDataset.getIncomingOperations(destination, 0, 11000);
+      outgoingOperations = fieldLineageDataset.getOutgoingOperations(source, 0, 11000);
+      // since checksum is same both for operations it should still store as same row
+      Assert.assertEquals(1, incomingOperations.size());
+      Assert.assertEquals(incomingOperations, outgoingOperations);
+      programRunOperations = incomingOperations.iterator().next();
+      Assert.assertEquals(new HashSet<>(Arrays.asList(programRun1, programRun2)),
+                          programRunOperations.getProgramRunIds());
+
+      // test for multiple set of operations
+      incomingOperations = fieldLineageDataset.getIncomingOperations(destination, 0, 12000);
+      outgoingOperations = fieldLineageDataset.getOutgoingOperations(source, 0, 12000);
+
+      Assert.assertEquals(2, incomingOperations.size());
+      Assert.assertEquals(incomingOperations, outgoingOperations);
+
+      Set<ProgramRunOperations> expectedSet = new HashSet<>();
+      expectedSet.add(new ProgramRunOperations(fieldLineageInfo.getChecksum(),
+              new HashSet<>(Arrays.asList(programRun1, programRun2)), fieldLineageInfo.getOperations()));
+      expectedSet.add(new ProgramRunOperations(anotherFieldLineageInfo.getChecksum(),
+              Collections.singleton(programRun3), anotherFieldLineageInfo.getOperations()));
+
+      Assert.assertEquals(expectedSet, incomingOperations);
+      Assert.assertEquals(expectedSet, outgoingOperations);
+    });
+  }
+
+  private static FieldLineageDataset getFieldLineageDataset(String instanceId) throws Exception {
+    DatasetId id = DatasetFrameworkTestUtil.NAMESPACE_ID.dataset(instanceId);
+    return DatasetsUtil.getOrCreateDataset(dsFrameworkUtil.getFramework(), id, FieldLineageDataset.class.getName(),
+                                           DatasetProperties.EMPTY, null);
+  }
+
+  private List<Operation> getOperations(String[] readOutput) {
```

Comment:
this is more a generateOperations() than a getter

Reply:
renamed.